### PR TITLE
Add SampleTypeRequestServiceTest

### LINKED ISSUE #3643: test(sample-type-request): add SampleTypeRequestService integration tests
Requires test DB + Liquibase table `clinlims.sample_type_request` (changeset `034-create-sample-type-request-table`).

## Out of scope

- No changes to `SampleTypeRequestDAOImpl` (separate bug issue).
- No new REST controller tests (existing `SampleTypeRequestRestControllerTest` stays Mockito-based).
- Does not modify shared `sampleitem.xml` (dedicated fixture only).

## Related

- DAO bug: **#3642** — `SampleTypeRequestDAOImpl` HQL `sampleId` type mismatch

### PR DESCRIPTION
**Pull request by Agaba-derrick**
# Pull Requests Requirements

- [x] The PR title includes a brief description of the work done, including the
      Issue number if applicable.
- [ ] The PR includes a video showing the changes for the work done.
- [ ] The PR title follows conventional commit label standards.
- [x] The changes confirm to the OpenElis Global x3
      [Styleguide and Design](https://uwdigi.atlassian.net/wiki/spaces/OG/pages/621346838/OpenELIS+Global+Style+Guide)
      documentation.
- [x] The changes include tests or are validated by existing tests.
- [x] I have read and agree to the Contributing
      [Guidelines](https://openelis-global.org/community/get-involved/) of this
      project.

## Summary
Adds comprehensive integration tests for `SampleTypeRequestService` and `SampleTypeRequestServiceImpl` using a real Spring context and database-backed fixtures.

### What was added

- Full CRUD integration test coverage for `SampleTypeRequest`
- Tests for request lifecycle transitions:
  - REQUESTED
  - COLLECTED
  - CANCELLED
- Coverage for:
  - request fulfillment
  - cancellation workflows
  - ordering and filtering queries
  - persistence operations
  - invalid state handling
  - exception scenarios
- Fixture validation tests for dataset consistency and ordering behavior
- Real database interaction testing with DBUnit datasets and Spring-managed transactions
- Verification of entity relationships between:
  - `Sample`
  - `SampleTypeRequest`
  - `TypeOfSample`
  - `SampleItem`

### Additional improvements

- Added reusable helper methods to reduce test duplication
- Expanded negative-path and edge-case coverage
- Ensured tests run without mocks to validate real Hibernate mappings and persistence behavior
## Screenshots
Screenshot from 2026-05-22 15-51-46

## Related Issue
Closes #3643

### CODE REVIEW

**Review by pmanko — APPROVED** [review]
Approving — real integration test against a Spring context + DBUnit, no mocks, good negative-path coverage. Two non-blocking notes:

- `getAllOrdered_shouldSortBySortOrderAscending` passes vacuously on a 0–1 row result; only the 4-row fixture makes the neighbor-comparison meaningful. Assert the explicit id order, like `fixture_shouldOrderBySortOrderWithinSample` already does.
- The `unit_of_measure` fixture row isn't referenced by any request, so its lazy-load path is never exercised — drop it or wire one up.

Beyond this PR: your Javadoc notes `getRequestsBySampleId` binds an `Integer` against `Sample`'s `String` PK — that's a real DAO bug, so scoping the tests around it here was the right call. Worth a follow-up ticket to fix the binding.

**Comment by Agaba-derrick** [comment]
Thanks @pmanko I already opened up a follow up Ticket for that issue here; https://github.com/orgs/DIGI-UW/projects/13?pane=issue&itemId=193351832&issue=DIGI-UW%7COpenELIS-Global-2%7C3642

File: src/test/java/org/openelisglobal/sampletyperequest/SampleTypeRequestServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+ * {@code SampleTypeRequestDAOImpl}, which currently binds {@code sampleId} as
+ * {@code Integer} while {@code Sample.id} is a {@code String} in Hibernate — so
+ * those methods throw at runtime for normal ids. We only integration-test their
+ * null/blank early-return paths here; fixture and workflow behavior for a
+ * sample is asserted via {@code get} / {@code getAll} / {@code getAllMatching}
+ * instead of changing production code under test.
+ */
+public class SampleTypeRequestServiceTest extends BaseWebContextSensitiveTest {
+
+    private static final String SAMPLE_WITH_REQUESTS = "2";
```
Comment (HerbertYiga) [inline src/test/java/org/openelisglobal/sampletyperequest/SampleTypeRequestServiceTest.java]: lets have the SAMPLE_WITH_REQUESTS  and SAMPLE_WITH_CANCELLED_ONLY well defined in your data set than setting them here

Comment (pmanko) [reply src/test/java/org/openelisglobal/sampletyperequest/SampleTypeRequestServiceTest.java]: The latest push dropped the `SAMPLE_WITH_*` constants — the sample semantics now live in the fixture (sample id=2 carries the requested/collected rows, id=1 the cancelled-only row), and the test filters via `requestsForSample()`. @HerbertYiga does that resolve your comment?

**Comment by pmanko** [comment]
Non-blocking, fine to defer to the #3642 follow-up:

1. `getAllOrdered_shouldSortBySortOrderAscending` would still pass on an empty result — with the fixed fixture you can assert the explicit sorted sequence `[1, 1, 2, 3]` instead of the pairwise loop.
2. The `unit_of_measure` fixture row appears unused — drop it or reference it.
3. The fixture header comment references the 001-sample-management feature — copy-paste leftover, worth updating to describe this dataset.
